storage/command_queue_test.go

```diff
@@ -171,3 +171,19 @@ func TestCommandQueueClear(t *testing.T) {
 		t.Fatal("commands should finish when clearing queue")
 	}
 }
+
+// TestCommandQueueExclusiveEnd verifies that an end key is treated as
+// an exclusive end when GetWait calculates overlapping commands. Test
+// it by calling GetWait with a command whose start key is equal to
+// the end key of a previous command.
+func TestCommandQueueExclusiveEnd(t *testing.T) {
```

Comment:
Ok. Testing that for the GC queue should be easy if you hook into `TestingCommandFilter`, intercept the GC command and verify its key range. I don't think your change has an effect though, since it only affects routing - the `GC` command will touch the last key regardless of whether it was specified in the header's key range. So you can't write a test that tests anything useful, at which point I don't think it's worth it.

Would you mind pulling out the `CommandQueue` test into a second commit?